modules/travel_pay/app/services/travel_pay/appointments_service.rb

```diff
@@ -45,14 +45,17 @@ def find_by_date_time(date_string, appointments)
         Rails.logger.error(message: 'Invalid appointment time provided (appointment time cannot be nil).')
         raise ArgumentError, message: 'Invalid appointment time provided (appointment time cannot be nil).'
       elsif date_string.present?
-        parsed_date_time = DateTime.parse(date_string)
-
+        parsed_date_time = DateUtils.strip_timezone(date_string)
         appointments.find do |appt|
-          !appt['appointmentDateTime'].nil? &&
-            parsed_date_time == DateTime.parse(appt['appointmentDateTime'])
+          begin
+            parsed_appt_time = DateUtils.strip_timezone(appt['appointmentDateTime'])
+          rescue TravelPay::InvalidComparableError => e
+            Rails.logger.warn("#{e} Appointment Datetime was nil for appointment #{appt['id'][0...8]}")
+          end
+          !appt['appointmentDateTime'].nil? && parsed_date_time.eql?(parsed_appt_time)
         end
       end
-    rescue DateTime::Error => e
+    rescue ArgumentError => e
```

Comment:
Does the `nil` and `ArgumentError` handling have any utility here since there looks to be something similar in `try_parse_date`? Or perhaps vice versa if it makes more sense to have it here?

Reply:
Good point. I vice versa'd it.